extras/data/machinery/generate/utils.py

```diff
@@ -81,15 +81,34 @@ def dump_polarized(src_path, target):
         observables dictionary, corresponding to the runcard
 
     """
+
+    # We make a case distinction for several experiments:
     obs = obs_template.copy()
-    data = load(src_path, 0, ["x", "Q2"])
-    dict_kins = [
-        dict(x=d["x"]["mid"], y=d["y"]["mid"], Q2=d["Q2"]["mid"]) for d in data
-    ]
+    data = np.genfromtxt(src_path)
+
+    if (
+        "ATHENA" in target.parent.name
+        or "EIC" in target.parent.name
+        or "EIcC" in target.parent.name
+    ):
```

Comment:
same holds as for my comment on `heavyness` - let's drop these ifs to a "higher level", i.e. outside of this function into the caller; see my previous comment about the options (and the freedom you have) - if you insist on a switch here, you could do it by file type (but NOT by name) and don't change the default how data is read (because the new way is actually the minority)